### PR TITLE
add a label to namespaces created by e2e framework

### PR DESCRIPTION
**Pull request by mikedanese**
makes cleanup easire when using --delete-namespace=false

k delete ns -l e2e-framework

cc @gmarek

### CODE REVIEW

**Comment by mikedanese** [comment]
flake finder:
```bash
count=1
while go run hack/e2e.go \
  -v -test \
  --test_args='--ginkgo.focus=complex --delete-namespace=false' \
  --check_version_skew=false;
do
  count=$((count+1))
  echo "------------------ count=${count}"
  k delete ns -l e2e-framework
done

File: test/e2e/example_cluster_dns.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ var _ = Describe("[Example] ClusterDns", func() {
 		namespaces := []*api.Namespace{nil, nil}
 		for i := range namespaces {
 			var err error
-			namespaces[i], err = createTestingNS(fmt.Sprintf("dnsexample%d", i), c)
+			namespaces[i], err = createTestingNS(fmt.Sprintf("dnsexample%d", i), c, nil)
```
Comment (ixdy) [inline test/e2e/example_cluster_dns.go]: is there a reason we don't want all of the namespaces created by e2e tests to have the label applied?

Comment (mikedanese) [inline test/e2e/example_cluster_dns.go]: now all test will have e2e-run label

**Comment by gmarek** [comment]
@k8s-bot unit test this

@mikedanese thanks!

**Comment by mikedanese** [comment]
@ixdy fixed typo!

**Comment by ixdy** [comment]
LGTM!

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by ixdy** [comment]
@k8s-bot e2e test this please

**Comment by gmarek** [comment]
@k8s-bot test this

**Comment by gmarek** [comment]
Failed to start cluster again...

@k8s-bot test this

**Pull request merged**